### DISCUSSION TITLE
Auth event for `USER_UPDATED` is not triggering

### DISCUSSION THREAD

**Discussion by eRuaro** [supabase/supabase#18795] [Questions]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I'm writing a reset password flow in React (just React, not Next) where I have a `forgot-password` page where users can input their email which will then send over an email containing the link to `reset-password`. `reset-password` is logging the `PASSWORD_RECOVERY` event but not the `USER_UPDATED` even though the password is updating.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Code for auth event listener:
```
useEffect(() => {
  // Define an event listener for authentication state changes
  const { data: authListener } = supabase.auth.onAuthStateChange(async (event, session) => {
    const user = await supabase.auth.getUser();

    console.log(event);

    if (event === "PASSWORD_RECOVERY") {
      console.log("Password recovery event");
      console.log("User email:", user.data.user.email);
    } else if (event === "USER_UPDATED") {
      console.log("User updated event");
      console.log("User email:", user.data.user.email);
      console.log("Session information:", session);
    }
  });

  // Cleanup function: unsubscribe from the event listener
  return () => {
    authListener.subscription.unsubscribe();
  };
}, []);
```

2. Code for onPasswordSubmit (when the user types in their new password):
```
const onPasswordSubmit = async (e) => {
        e.preventDefault();

        console.log("Password update");
        console.log(password);
        console.log(confirmPassword);

        if (password !== confirmPassword) {
            setErrorMessage("Passwords do not match");
            setErrorDescription("Please make sure your passwords match");
            setShowErrorNotification(true);
            return;
        }

        const { data, error } = await supabase.auth.updateUser({
            password: password,
        });
        
        if (data) {
            setUserNotificationMessage("Password updated");
            setUserNotificationDescription("Your password has been updated successfully, you can now login with your new password.");
            setShowUserNotification(true);
        }

        if (error) {
            setErrorMessage("Error updating password");
            setErrorDescription(error.message);
            setShowErrorNotification(true);
    
    }
    };
```

## Expected behavior
The thing is though, it's not triggering an `USER_UPDATED` which is what I'm expecting it to do after this code. Plus, it does not show the notifications when the password is updated, however, upon looking at network logs, and logging into the app, the password was in fact updated.

## System information

- OS: [Ubuntu (WSL)]
- Browser (if applies) [e.g. chrome, firefox]
- Version of supabase-js: [2.36]
- Version of Node.js: [19.6]

**Comment by eRuaro** [comment]
Fixed the issue by apparently removing the `useEffect` altogether although I suspect this isn't how it should be done?

**Comment by montassar-vaionex** [comment]


**Reply by GaryAustin1** [reply]
That is a server call and I don't see how the event handler for browser events will know it occurred.
The docs only say it will trigger on the plain auth.updateUser() call.
![image](https://github.com/supabase/supabase/assets/54564956/bc62c743-683b-41a1-899c-f4790296a60e)
